services/supervisord/supervisord.go

```diff
@@ -679,4 +680,42 @@ stdout_logfile_maxbytes = 10MB
 stdout_logfile_backups = 3
 redirect_stderr = true
 {{end}}
+
+{{define "grafana"}}
+[program:grafana]
+priority = 3
+command =
+    /usr/sbin/grafana-server
+        --homepath=/usr/share/grafana
+        --config=/etc/grafana/grafana.ini
+        cfg:default.paths.data=/srv/grafana
+        cfg:default.paths.plugins=/srv/grafana/plugins
+        cfg:default.paths.logs=/srv/logs
+        cfg:default.log.mode=console
+        cfg:default.log.console.format=console
+        cfg:default.server.root_url="https://%%(domain)s/graph"
+        {{- if .PerconaSSODetails}}
+        cfg:default.auth.generic_oauth.enabled=true
+        cfg:default.auth.generic_oauth.name="Percona Account"
+        cfg:default.auth.generic_oauth.client_id="{{ .PerconaSSODetails.ClientID }}"
+        cfg:default.auth.generic_oauth.client_secret="{{ .PerconaSSODetails.ClientSecret }}"
+        cfg:default.auth.generic_oauth.scopes="openid profile email offline_access"
+        cfg:default.auth.generic_oauth.auth_url="{{ .PerconaSSODetails.IssuerURL }}/authorize"
+        cfg:default.auth.generic_oauth.token_url="{{ .PerconaSSODetails.IssuerURL }}/token"
+        cfg:default.auth.generic_oauth.api_url="{{ .PerconaSSODetails.IssuerURL }}/userinfo"
+        {{- end}}
```

Comment:
Why do we need this data here? How grafana uses it?

Reply:
This is to enable single sign on.

Reply:
the tech spec mentions setting the sso params in `grafana.ini` before re-starting grafana https://confluence.percona.com/display/PMM/Single+Sign-On+-+PMM+Server+integration+with+Portal+and+Okta

see point 8

Reply:
Yes, but we would need to solve conflicts because it gets modified also during PMM update. So we decided to go with this.

Reply:
please reflect these changes in the spec later